graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/MappedFieldTypeDTO.java

```diff
@@ -17,22 +17,20 @@
 package org.graylog.plugins.views.search.rest;
 
 import com.fasterxml.jackson.annotation.JsonAutoDetect;
-import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.google.auto.value.AutoValue;
+import org.graylog.plugins.formatting.units.model.UnitId;
 import org.graylog2.indexer.fieldtypes.FieldTypes;
 
-@AutoValue
-@JsonAutoDetect
-public abstract class MappedFieldTypeDTO {
-    @JsonProperty("name")
-    public abstract String name();
+import javax.annotation.Nullable;
 
-    @JsonProperty("type")
-    public abstract FieldTypes.Type type();
+@JsonAutoDetect
+public record MappedFieldTypeDTO(@JsonProperty("name") String name,
+                                 @JsonProperty("type") FieldTypes.Type type,
+                                 @JsonProperty("unit") @Nullable UnitId unit
+) {
 
-    @JsonCreator
-    public static MappedFieldTypeDTO create(@JsonProperty("name") String name, @JsonProperty("type") FieldTypes.Type type) {
-        return new AutoValue_MappedFieldTypeDTO(name, type);
+    public MappedFieldTypeDTO(final String name,
```

Comment:
This change is unnecessary and provokes a lot of fallout. We should keep the static factory method and avoid the changes required by this.